src/utils.ts

```diff
@@ -260,3 +262,45 @@ export function asArray<T>(value: T | T[]): T[] {
 export type Dictionary<T = unknown> = Record<PropertyKey, T>;
 
 export type DistributiveOptional<T, K extends keyof T> = T extends any ? Omit<T, K> & Partial<Pick<T, K>> : never;
+
+/**
+ * Creates a storage signature for a resource, which can be used to generate signed URLs for accessing the resource.
+ * The signature is created using HMAC with the provided secret key and includes the resource ID, expiration time, and version.
+ *
+ * Note: expirationMillis is optional. If not provided, the signature will not expire.
+ */
+export function createStorageSignature({
+    resourceId,
+    urlSigningSecretKey,
+    expiresInMillis,
+    version = 0,
+}: {
+    resourceId: string;
+    urlSigningSecretKey: string;
+    expiresInMillis: number | undefined;
+    version?: number;
+}) {
+    const expiresAt = expiresInMillis ? new Date().getTime() + expiresInMillis : null;
```

Comment:
```suggestion
    const expiresAt = expiresInMillis ? new Date().getTime() + expiresInMillis : '0';
```

(opt) Maybe this? 

Then in API you'll be able to do:

```typescript
const parsedExpiresAtMillis = parseInt(expiresAtMillis, 10);
if (parsedExpiresAtMillis > 0 && parsedExpiresAtMillis < new Date().getTime()) {
    // signature expired
    return false;
}

Reply:
Honestly, I'd keep it as `null, but I updated comments in API 👇 

```
// For non-expiring signatures, we pass 'null' as timestamp.
// If the timestamp is 'null', we don't need to check for expiration.
if (expiresAtMillis !== 'null' && parseInt(expiresAtMillis, 10) < new Date().getTime()) {
  // signature expired
  return false;
}
```

Reply:
well, but it is not required to check for `expiresAtMillis !== 'null'`, I thought it would be more self-explanatory 😄 

`NaN < new Date().getTime()` would always return false

Reply:
Okay, you're right 😄
 
In python it will looks ugly, since python has `None` not `null`.

`expires_at_str = str(expires_at) if expires_at is not None else 'null'`

Changing it to 0